xbmc/pvr/recordings/PVRRecordings.cpp

```diff
@@ -194,17 +196,28 @@ int CPVRRecordings::GetNumRecordings()
   return m_recordings.size();
 }
 
-int CPVRRecordings::GetRecordings(CFileItemList* results)
+bool CPVRRecordings::HasDeletedRecordings()
+{
+  CSingleLock lock(m_critSection);
+  return m_bHasDeleted;
+}
+
+int CPVRRecordings::GetRecordings(CFileItemList* results, bool deleted)
```

Comment:
then we can also simplify the way you now determine whether there are deleted recordings (extra state boolean m_bHasDeleted vs. checking if the vector with deleted recordings is empty).